packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/dropNode/DropNodeContext.tsx

```diff
@@ -12,23 +12,49 @@
  *******************************************************************************/
 
 import React, { useState } from 'react';
-import { DropNodeContextProviderProps, DropNodeContextValue } from './DropNodeContext.types';
-import { NodeDropData } from './useDropNode.types';
+import { DropNodeContextProviderProps, DropNodeContextValue, NodeDropData } from './DropNodeContext.types';
 
 const defaultValue: DropNodeContextValue = {
-  dropData: {
-    initialParentId: null,
-    draggedNodeId: null,
-    targetNodeId: null,
-    compatibleNodeIds: [],
-    droppableOnDiagram: false,
-  },
+  initialParentId: null,
+  draggedNode: null,
+  targetNodeId: null,
+  compatibleNodeIds: [],
+  droppableOnDiagram: false,
   setDropData: () => {},
+  setTargetNodeId: () => {},
 };
 
 export const DropNodeContext = React.createContext<DropNodeContextValue>(defaultValue);
 
 export const DropNodeContextProvider = ({ children }: DropNodeContextProviderProps) => {
-  const [dropData, setDropData] = useState<NodeDropData>(defaultValue.dropData);
-  return <DropNodeContext.Provider value={{ dropData, setDropData }}>{children}</DropNodeContext.Provider>;
+  const [state, setState] = useState<NodeDropData>({
+    initialParentId: null,
+    draggedNode: null,
+    targetNodeId: null,
+    compatibleNodeIds: [],
+    droppableOnDiagram: false,
+  });
+
+  const setDropData = (dropData: NodeDropData) => {
+    setState((prevState) => ({ ...prevState, ...dropData }));
+  };
+
+  const setTargetNodeId = (targetNodeId: string | null) => {
+    setState((prevState) => ({ ...prevState, targetNodeId }));
+  };
```

Comment:
Why both a function to change the whole state and one to change one specific part of the state?

Reply:
Yes, as you mentioned that, I should have defined three functions:
- `initDrop` instead of `setDropData`
- `updateTargetNodeId` (or keep `setTargetNodeId`)
- `resetDrop` or something like that